### ISSUE TITLE
error: dereferencing type-punned pointer will break strict-aliasing rules

### ISSUE THREAD

**Issue by matthewbauer** [facebook/xcbuild#221]
We are getting the following warning (being treated as an error) while building xcbuild with GCC compile:

```
/tmp/nix-build-xcbuild-0.1.1.drv-0/xcbuild-0.1.1-src/Libraries/plist/Sources/Format/ABPReader.cpp: In member function 'plist::Real* ABPReader::readReal(size_t)':
/tmp/nix-build-xcbuild-0.1.1.drv-0/xcbuild-0.1.1-src/Libraries/plist/Sources/Format/ABPReader.cpp:111:75: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
         case 4:  return plist::Real::New(*reinterpret_cast<float *>(&value)).release();
                                                                           ^
/tmp/nix-build-xcbuild-0.1.1.drv-0/xcbuild-0.1.1-src/Libraries/plist/Sources/Format/ABPReader.cpp:112:76: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
         case 8:  return plist::Real::New(*reinterpret_cast<double *>(&value)).release();
                                                                            ^
cc1plus: all warnings being treated as errors
make[2]: *** [Libraries/plist/CMakeFiles/plist.dir/build.make:687: Libraries/plist/CMakeFiles/plist.dir/Sources/Format/ABPReader.cpp.o] Error 1
```

Adding the flag -Wno-error=strict-aliasing fixes it.

**Comment by grp** [comment]
@strager, do you know the best way to fix this?

**Comment by strager** [comment]
Use memcpy instead of reinterpret_cast. memcpy (used in this way) results implementation-defined behaviour, but that's probably okay for now.

**Comment by strager** [comment]
(The reinterpret_cast solution is undefined behaviour, so memcpy is strictly better.)

**Comment by grp** [comment]
Fixed with #222.
